lib/puma/cluster.rb

```diff
@@ -267,6 +282,8 @@ def worker(index, master)
       @launcher.config.run_hooks :before_worker_boot, index
 
       server = start_server
+      server.update_capacity = Proc.new {|c| @worker_write << "!c#{Process.pid},#{c}\n" }
```

Comment:
Note there is a bit of synchronization overhead in this operation, which starts to become visible at extremely low-latency response times / high request rates (e.g., ~1ms/req, ~10k req/sec). An alternative is to use shared memory (e.g. the [raindrops](https://bogomips.org/raindrops/) gem), see 486b4f9206767c1e6f08b6697b2d8695246c6629 for a commit implementing this alternative approach. The tradeoff is pulling in an extra dependency (or writing extra C-extension code) for that extra bit of performance.